### PR TITLE
fix(build): Escape dollar signs in dart-transpiled string literals

### LINKED ISSUE #509: [transpiler] Wrong handling of string templates
In Dart string interpolation works regardless of quote style. In ES6 string templates perform interpolation and old-style strings do not. The transpiler is unaware of that and therefore produces wrong result.

This:
```
var a = 1;
var s1 = '${a}';
var s2 = `${a}`;
```

Transpiles to this:

```
var a = 1;
var s1 = '${a}';
var s2 = '''${a}''';
```

Which is wrong. I believe this currently results in a bug with `$event` in DOM event binding.

### PR DESCRIPTION
**Pull request by alexeagle**
Escape dollar signs in string literals - dart should not interpolate them.
Closes #509

### CODE REVIEW

File: tools/transpiler/src/outputgeneration/DartParseTreeWriter.js [hunk 1]
```diff
@@ -444,6 +444,12 @@ export class DartParseTreeWriter extends JavaScriptParseTreeWriter {
     super.visitArrayLiteralExpression(tree);
   }
 
+  visitLiteralExpression(tree) {
```
Comment (yjbanov) [inline tools/transpiler/src/outputgeneration/DartParseTreeWriter.js]: `visitTemplateLiteralPortion` also needs to be fixed

File: tools/transpiler/unittest/transpilertests.js [hunk 2]
```diff
@@ -15,13 +16,14 @@ describe('transpile to dart', function(){
   // https://github.com/angular/angular/issues/509
   it('should not interpolate inside old quotes', function(){
     var result = compiler.compile(OPTIONS, "test.js",
-      "var a = 1;" +
-      "var s1 = '${a}';" +
-      "var s2 = `${a}`;");
+      "var a:number = 1;" +
+      "var s1:string = '${a}';" +
+      "var s2:string = `${a}`;" +
```
Comment (yjbanov) [inline tools/transpiler/unittest/transpilertests.js]: Also include tests without curlies, `'$a'` etc

Comment (alexeagle) [inline tools/transpiler/unittest/transpilertests.js]: Added more test cases (which were broken) and fixed them.

**Comment by yjbanov** [comment]
lgtm

**Pull request closed without merging**